lib/src/analysis_options.dart

```diff
@@ -13,8 +16,80 @@ linter:
   rules:
     - camel_case_types
     - hash_and_equals
+    # TODO: flutter doesn't use it, make it consistent
     - iterable_contains_unrelated_type
+    # TODO: flutter doesn't use it, make it consistent
     - list_remove_unrelated_type
     - unrelated_type_equality_checks
     - valid_regexps
 ''';
+
+// Keep it updated with
+// https://github.com/flutter/flutter/blob/master/packages/flutter/lib/analysis_options_user.yaml
+const String flutterAnalysisOptions = '''
+analyzer:
+  language:
+    enableStrictCallChecks: true
+    enableSuperMixins: true
+    enableAssertInitializer: true
+  strong-mode: true
+  errors:
+    # treat missing required parameters as a warning (not a hint)
+    missing_required_param: warning
+    # treat missing returns as a warning (not a hint)
+    missing_return: warning
+    # allow having TODOs in the code
+    todo: ignore
+
+linter:
```

Comment:
Let's remove these. return types on setters, use isNotEmpty, etc have no affect on runtime.

Let's use one set of lints and just turn on the language, strong-mode, and error tweaks

Reply:
Like this? or do you want to have the error tweaks in the default options too?

Reply:
Yup!

We could even share the lints in a variable...but not a biggy.

Reply:
yeah. the benefit of this version: easy to copy-paste. Can do the variable thing if you prefer that.

Reply:
No biggy